base/lock.jl

```diff
@@ -570,3 +577,274 @@ end
     import .Base: Event
     export Event
 end
+
+const PerStateInitial       = 0x00
+const PerStateHasrun        = 0x01
+const PerStateErrored       = 0x02
+const PerStateConcurrent    = 0x03
+
+"""
+    OncePerProcess{T}
+
+Calling a `OncePerProcess` object returns a value of type `T` by running the
+function `initializer` exactly once per process. All concurrent and future
+calls in the same process will return exactly the same value. This is useful in
+code that will be precompiled, as it allows setting up caches or other state
+which won't get serialized.
+
+## Example
+
+```jldoctest
+julia> const global_state = Base.OncePerProcess{Vector{UInt32}}() do
+           println("Making lazy global value...done.")
+           return [Libc.rand()]
+       end;
```

Comment:
```suggestion
       end
...
```

Showing the return type here may be helpful. That will show us that `global_state isa Base.OncePerProcess`.

Reply:
That would print a lot of meaningless details, since I don't override show (or deepcopy or serialize for that matter), as I want to show the behavior, not the name of the return type